pandas/core/indexes/base.py

```diff
@@ -3842,20 +3842,9 @@ def _convert_slice_indexer(self, key: slice, kind: Literal["loc", "getitem"]):
             # We always treat __getitem__ slicing as label-based
             # translate to locations
             if kind == "getitem" and is_index_slice and not start == stop and step != 0:
-                # exclude step=0 from the warning because it will raise anyway
-                # start/stop both None e.g. [:] or [::-1] won't change.
-                # exclude start==stop since it will be empty either way, or
-                # will be [:] or [::-1] which won't change
-                warnings.warn(
-                    # GH#49612
-                    "The behavior of obj[i:j] with a float-dtype index is "
-                    "deprecated. In a future version, this will be treated as "
-                    "positional instead of label-based. For label-based slicing, "
-                    "use obj.loc[i:j] instead",
-                    FutureWarning,
-                    stacklevel=find_stack_level(),
+                raise TypeError(
```

Comment:
It’s not unsupported. The point is to change the behavior to not special-case here

Reply:
Thanks for the comment. I corrected my mistake, fixed tests, and added a note to v3.0.0

Reply:
@jbrockmendel could you please take a look at my changes?